tests/object_store.cpp

```diff
@@ -30,3 +37,102 @@ TEST_CASE("ObjectStore: table_name_for_object_type()") {
         REQUIRE(result == "class_good");
     }
 }
+
+TEST_CASE("ObjectStore:: property_for_column_index()") {
+    SECTION("Property should match the schema") {
+        Schema schema = {
+            {"object", {
+                {"int", PropertyType::Int},
+                {"boolNullable", PropertyType::Bool | PropertyType::Nullable},
+                {"stringPK", PropertyType::String, true},
+                {"dateNullableIndexed", PropertyType::Date | PropertyType::Nullable, false, true},
+                {"floatNullableArray", PropertyType::Float | PropertyType::Nullable | PropertyType::Array},
+                {"doubleArray", PropertyType::Double | PropertyType::Array},
+                {"object", PropertyType::Object | PropertyType::Nullable, "object"},
+                {"objectArray", PropertyType::Object | PropertyType::Array, "object"},
+            }}
+        };
+
+        TestFile config;
+        config.schema = schema;
+        config.schema_version = 1;
+
+        auto realm = Realm::get_shared_realm(config);
+        ConstTableRef table = ObjectStore::table_for_object_type(realm->read_group(), "object");
+
+        size_t count = table->get_column_count();
+        for (size_t col = 0; col < count; col++) {
+            auto property = ObjectStore::property_for_column_index(table, col);
+            if (!property) {
+#if REALM_HAVE_SYNC_STABLE_IDS
+                REQUIRE(table->get_column_name(col) == sync::object_id_column_name);
+#else
+                FAIL();
+#endif
+                continue;
+            }
+
+            REQUIRE(property->name == table->get_column_name(col));
+            if (property->name == "int") {
+                REQUIRE(property->type == PropertyType::Int);
```

Comment:
Rather than hard-coding these various tests, could we instead look up the expected `Property` in `schema` and use the overload of `operator==` to compare expected vs actual?

Reply:
Yes! much less code